iscsi/iscsi.go

```diff
@@ -259,14 +262,14 @@ func Connect(c Connector) (string, error) {
 
 		// create our devicePath that we'll be looking for based on the transport being used
 		if c.Port != "" {
-			p = strings.Join([]string{p, c.Port}, ":")
+			target.Portal = strings.Join([]string{target.Portal, c.Port}, ":")
```

Comment:
This is where I wonder if that port option should be folded up under the same targets struct you created rather than only providing a single global option.  Maybe even just modify the portal inputs to include the port suffix, or if omitted obviously just use the default 3260